ios_list_data/Networking/APIHandler.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  APIHandler.swift
+//  ios_list_data
+//
+//  Created by Nam on 13/12/2022.
+//
+
+import Foundation
+import Alamofire
+
+class APIHandler{
+    static let sharedIntance = APIHandler()
+    
+    func fetchingAPIData(keyWord: String, time:String,sort: String?, completion:@escaping(Result<ListItemModel, Error>)-> Void){
+        print("fetchingAPIData")
+        let decoder : JSONDecoder = {
+            let decoder = JSONDecoder()
+            return decoder
+        }()
+        let url = "https://newsapi.org/v2/everything?q=\(keyWord)&from=\(time)&sortBy=\(sort ?? publishAt)&apiKey=\(API_KEY)"
```

Comment:
lets AF build request with param passed, by set urlEncoding
APIHandler do not need to know what is params of url request
create a struct: NewsRepository to build param and pass to APIHandler (should be a dictionary)
viewmodel -> repository -> apihandler